src/Magento/FunctionalTestingFramework/Test/Util/TestHookObjectExtractor.php

```diff
@@ -57,19 +57,22 @@ public function extractHook($parentName, $hookType, $testHook)
 
     /**
      * Creates the default failed hook object with a single saveScreenshot action.
+     * And a pause action when ENABLE_PAUSE is set to true.
      *
      * @param string $parentName
      * @return TestHookObject
      */
     public function createDefaultFailedHook($parentName)
     {
-
-        $saveScreenshotStep = [new ActionObject("saveScreenshot", "saveScreenshot", [])];
+        $defaultSteps['saveScreenshot'] = new ActionObject("saveScreenshot", "saveScreenshot", []);
```

Comment:
Modified AdminLoginSuccessfulTest to use an incorrect username, couldn't get the test to pause at the failed step. Can you double check?

Reply:
It requires `ENABLE_PAUSE` to be set. Did you have it on when run the test?

Reply:
yes I did have it set to true, the test just executed without pausing and did not open the interactive shell.

Reply:
The pause() will be added in the _failed() hook. You need to make sure the test has an after hook in the first place. This is an existing behavior I discovered during development and it seems that we agree on keeping the current behavior.
See here https://magento.slack.com/archives/GM9SE437U/p1596745949016600

Reply:
How about adding this `$this->pause()` to _failed() of MagentoWebDriver? I thought of it because saveScreenshot gets called for tests not having after block.

I added `$this->pause()` to _failed() method and it seemed to pause at failure and invoked interactive console

Reply:
I have some concern. Will it cause test to pause during execution? Does it change the generated code?

Reply:
I don't think it changes generated cest. It paused test if there is a failure encountered.

Reply:
I added `pause` with condition. I intentionally don't want to add any pause() for run time. If we decided to do it, I think there are other places too.

Reply:
do you mean not adding pause for codecept:run? Currently using `codecept:run --debug` without enable pause does not pause tests at failure. Will users expect it pause at failures?

Reply:
If you use `codecept:run --debug`, mftf should pause without the flag. Do you see differently?